metpy/deprecation.py

```diff
@@ -263,33 +263,23 @@ def deprecate(obj, message=message, name=name, alternative=alternative,
             func = obj.__init__
 
             def finalize(wrapper, new_doc):
-                try:
-                    pass
-                    # obj.__doc = new_doc
-                except (AttributeError, TypeError):
-                    # cls.__doc__ is not writeable on Py2.
-                    # TypeError occurs on PyPy
-                    pass
+                # try:
```

Comment:
I appreciate trying to improve the test coverage, but for this particular file, I'd like to minimize divergence from the original matplotlib code to the extent possible.

Reply:
So I looked into this and matplotlib's current deprecation code is even more matplotlib specific and has a new intra-code reference to `cbook._warn_external`.  Seems like there isn't much to gain by using matplotlib's machinery here?  shrug.

Reply:
Ok, then I guess we can take your mods. Can you remove the extra code rather than commenting it out?

Reply:
rebased and redid the first commit.